File: spec/cpp_stl_98/test_to_string_custom.cpp
#include <boost/test/unit_test.hpp>
#include "to_string_custom.h"
#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_to_string_custom) {
    std::ifstream ifs("src/term_strz.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    to_string_custom_t* r = new to_string_custom_t(&ks);

    std::stringstream ss;
    ss << *r;

    BOOST_CHECK_EQUAL(ss.str(), "s1 = foo, s2 = bar");

    delete r;
}

File: spec/python/spec/test_to_string_custom.py
import unittest
from testformats.to_string_custom import ToStringCustom

class TestToStringCustom(unittest.TestCase):
    def test_to_string_custom(self):
        with ToStringCustom.from_file('src/term_strz.bin') as r:
            self.assertEqual(str(r), "s1 = foo, s2 = bar")
